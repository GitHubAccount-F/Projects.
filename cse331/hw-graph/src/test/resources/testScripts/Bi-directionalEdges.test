#Tests to see if two nodes can have bi-directional edges(edges
#that goes both ways). We also check if this property
#holds for multiple edges going back and forth between two nodes

#Creates empty graphs
CreateGraph graph1
CreateGraph graph2
CreateGraph graph3

#Adds multiple nodes
AddNode graph1 one
AddNode graph1 two
AddNode graph2 one
AddNode graph2 two
AddNode graph3 one
AddNode graph3 two

#Creates multiple edges
AddEdge graph1 one two e1
AddEdge graph1 two one e1
AddEdge graph2 one two e1
AddEdge graph2 two one e1
AddEdge graph2 one two e2
AddEdge graph3 two one e1
AddEdge graph3 one two e1
AddEdge graph3 two one e2
AddEdge graph3 one two e2

#Checks to see if the bi-directional edges were added,
#where it creates child nodes in both directions
ListChildren graph1 one
ListChildren graph1 two
ListChildren graph2 one
ListChildren graph2 two
ListChildren graph3 one
ListChildren graph3 two