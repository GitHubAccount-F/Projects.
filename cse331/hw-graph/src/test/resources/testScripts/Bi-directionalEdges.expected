#Tests to see if two nodes can have bi-directional edges(edges
#that goes both ways). We also check if this property
#holds for multiple edges going back and forth between two nodes

#Creates empty graphs
created graph graph1
created graph graph2
created graph graph3

#Adds multiple nodes
added node one to graph1
added node two to graph1
added node one to graph2
added node two to graph2
added node one to graph3
added node two to graph3

#Creates multiple edges
added edge e1 from one to two in graph1
added edge e1 from two to one in graph1
added edge e1 from one to two in graph2
added edge e1 from two to one in graph2
added edge e2 from one to two in graph2
added edge e1 from two to one in graph3
added edge e1 from one to two in graph3
added edge e2 from two to one in graph3
added edge e2 from one to two in graph3

#Checks to see if the bi-directional edges were added,
#where it creates child nodes in both directions
the children of one in graph1 are: two(e1)
the children of two in graph1 are: one(e1)
the children of one in graph2 are: two(e1) two(e2)
the children of two in graph2 are: one(e1)
the children of one in graph3 are: two(e1) two(e2)
the children of two in graph3 are: one(e1) one(e2)